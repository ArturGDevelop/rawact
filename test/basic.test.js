import React from "react";
import ReactDOM from "react-dom";
import * as Basic from './cases/basic';

describe('basic', () => {
	it('renders correctly', () => {
		const element = document.createElement('div');

		const elementRendered = ReactDOM.render(<Basic.HelloWorld/>, element);

		expect(element).toMatchSnapshot();
		expect(elementRendered).toEqual(element.childNodes[0]);
	});

	it('renders updates from render', () => {
		const element = document.createElement('div');

		ReactDOM.render(<Basic.Hello name="testing"/>, element);
		ReactDOM.render(<Basic.Hello name="world"/>, element);

		expect(element).toMatchSnapshot();
	});
});
