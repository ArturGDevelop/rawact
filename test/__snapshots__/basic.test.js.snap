// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic renders correctly 1`] = `
<div>
  <div
    title="hello"
  >
    <p>
      Hello 
      World
      !
    </p>
    !
    !
  </div>
</div>
`;

exports[`basic renders updates from outside 1`] = `
<div>
  <p>
    Hello 
    world
    !
  </p>
</div>
`;
